packages/openapi-generator/src/generator.ts

```diff
@@ -74,94 +93,55 @@ export async function generate(options: GeneratorOptions): Promise<void> {
       ? 'Some clients could not be generated.'
       : 'Could not generate client.';
   await finishAll(promises, errorMessage);
+
+  if (options.perServiceConfig) {
+    writeFile(
```

Comment:
I am confused. The getPerServiceConfig just reads it and than we write it here. Where is the point where it is initialized with default values if it is not there?

Reply:
Ah, part of the confusion is related to the fact that the `getPerServiceConfig` return a mapping for all services. I did not get this at first. Perhaps you can adjust the name to be plural?

Reply:
or use a list of service configs and use a find to get the right one in the getOrCreateServiceConfig()

Reply:
A list is problematic, because we never need it in this format. It is currently a map from file name to `ServiceConfig`. As the parameter is called `perServiceConfig` it only makes sense to give the structure a name that relates to that.
I understand the confusion, but the plural of "configuration" is "configuration", so I am not sure how to improve that.

What I can do though is set the config explicitly. It will make the core more complex, but more explicit. I'll check what this means in practice and either just do it, or come back with questions.

Reply:
> What I can do though is set the config explicitly. It will make the core more complex, but more explicit. I'll check what this means in practice and either just do it, or come back with questions.

I do not really understand what you mean with that but let's see.

Reply:
Always create the object in the code, always get the service config and in the end only write if the parameter was set.

Reply:
Rename: 
`openapi-generator --optionsPerService`
`OptionsPerService`
`ServiceOptions`

+ update ADR

Reply:
check